pkg/syncer/spec/dns/dns_process.go

```diff
@@ -93,6 +101,31 @@ func (d *DNSProcessor) EnsureDNSUpAndReady(ctx context.Context, workspace logica
 	logger.V(4).Info("checking if all dns objects exist and are up-to-date")
 	ctx = klog.NewContext(ctx, logger)
 
+	// Try updating resources if not done already
+	if d.updating.Load().(bool) {
+		d.updatingMu.Lock()
```

Comment:
Do you think you could factorize this inside a func, so that you can use a `defer` call to unlock the mutex ?

Reply:
done